geniza/corpus/models.py

```diff
@@ -925,6 +925,65 @@ def title(self):
         return f"{self.doctype or _('Unknown type')}: {self.shelfmark_display or '??'}"
 
+    def dating_range(self):
+        """
+        Return the start and end of the document's possible date range, as PartialDate objects,
+        including standardized document dates and inferred Datings, if any exist.
+        """
+        # it is unlikely, but technically possible, that a document could have both on-document
+        # dates and inferred datings, so find the min and max out of all of them.
+        start_date = self.start_date if self.start_date else None
+        end_date = self.end_date if self.end_date else None
+        dating_range = [start_date, end_date]
+
+        # bail out if we don't have any inferred datings
+        if not self.dating_set.exists():
+            return tuple(dating_range)
+
+        # loop through inferred datings to find min and max
+        for dating in self.dating_set.all():
+            # get start from standardized date range (formatted as "date1/date2" or "date")
+            split_date = dating.standard_date.split("/")
+            start = PartialDate(split_date[0])
+            # use numeric format to compare to current min, replace if smaller
+            start_numeric = int(start.numeric_format(mode="min"))
+            if dating_range[0] is None or start_numeric < int(
+                dating_range[0].numeric_format(mode="min")
+            ):
+                dating_range[0] = start
+            # get end from standardized date range
+            end = PartialDate(split_date[1]) if len(split_date) > 1 else start
+            # use numeric format to compare to current max, replace if larger
+            end_numeric = int(end.numeric_format(mode="max"))
+            if dating_range[1] is None or end_numeric > int(
+                dating_range[1].numeric_format(mode="max")
+            ):
+                dating_range[1] = end
+
+        return tuple(dating_range)
+
+    def solr_dating_range(self):
+        """Return the document's dating range, including inferred, as a Solr date range."""
+        dating_range = self.dating_range()
+        solr_dating_range = []
+        if dating_range[0] is not None:
+            # min from dating_range, formatted YYYY-MM-DD or YYYY-MM or YYYY
+            solr_dating_range.append(dating_range[0].isoformat())
+        # if there's a max date in the range, ensure it's not the same as the min
+        if (
+            dating_range[1] is not None
+            and dating_range[1].isoformat(mode="max") != solr_dating_range[0]
+        ):
+            # max from dating_range, formatted YYYY-MM-DD or YYYY-MM or YYYY
+            solr_dating_range.append(dating_range[1].isoformat(mode="max"))
+        if not solr_dating_range:
+            return None
+        # if a single date instead of a range, start and end are the same
+        if len(solr_dating_range) == 1:
+            return solr_dating_range[0]
```

Comment:
is this equivalent?  (white space is probably not right)

```suggestion
        solr_dating_range = []
         for i, date in enumerate(self.dating_range()):
             if date: 
                # min/max from dating_range, formatted YYYY-MM-DD or YYYY-MM or YYYY
                solr_dating_range.append(date.isoformat(mode='max' if i == 1 else 'min'))
        if not solr_dating_range:
            return None
        # if a single date instead of a range, start and end are the same
        if solr_dating_range[0] == solr_dating_range[1]:
            return solr_dating_range[0]
```

Reply:
Yes, great simplification!